### PR TITLE
Refactor acceptance tests

### PR DESCRIPTION
**Pull request by jonaszkadziela**
### Refactored automated tests
* Converted some pieces out of unreasonably long feature specs into view & request specs
* Added `:filled_with_content` traits to profile & section factories
* Added `id` to the `div` of profile management panel so as to avoid future issues
* Adjusted id names to Rails convention (words separated by underscores). Earlier ids had camel case names

### CODE REVIEW

File: spec/factories/profile.rb [hunk 1]
```diff
@@ -2,5 +2,13 @@
   factory :profile do
     sequence(:name) { |n| "Profile name #{n}" }
     user
+
+    trait :filled_with_content do
+      after(:create) do |profile|
+        [*1..5].sample.times do
```
Comment (PiotrMisiurek) [inline spec/factories/profile.rb]: I would recommend using a fixed number of sections here. In the longer period, you or another developer can write some spaces and has WTF? why sometimes I have one section, but another time I have more of them. 

Our goal is to write specs that always behave the same way. Adding this kind of randomness can cause randomly failing specs in long-term

BTW very nice use of `[*1..5]`, i didn't know that way of conversion range to array 👍

File: spec/factories/section.rb [hunk 2]
```diff
@@ -2,5 +2,13 @@
   factory :section do
     sequence(:name) { |n| "Section name #{n}" }
     profile
+
+    trait :filled_with_content do
+      after(:create) do |section|
+        [*1..5].sample.times do
```
Comment (PiotrMisiurek) [inline spec/factories/section.rb]: The same as about, we don't want randomness in our specs

File: spec/views/profiles/index.html.haml_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      render template: "profiles/index.html.haml"
+
+      expect(rendered).to have_link(profile1.name)
+      expect(rendered).to have_link(profile2.name)
+    end
+  end
+
+  context "when there are no profiles" do
+    it "shows an information about no profiles to list" do
+      Profile.delete_all
```
Comment (PiotrMisiurek) [inline spec/views/profiles/index.html.haml_spec.rb]: Not the best solution. You create profiles in lines 4 and 5 just to delete them here. We don't need to do this db calls (btw good that you use `delete_all` to reduce db overhead.

I would recommend moving `lets` to `when there is at least one profile context` as an optimal solution here

**Review by kubi-dev — COMMENTED** [review]
It's going in good direction. Good job

File: spec/factories/section.rb [hunk 4]
```diff
@@ -2,5 +2,13 @@
   factory :section do
     sequence(:name) { |n| "Section name #{n}" }
     profile
+
+    trait :filled_with_content do
+      after(:create) do |section|
+        4.times do
+          create(:item, Item::TYPES.sample, section: section)
```
Comment (PiotrMisiurek) [inline spec/factories/section.rb]: `sample` add randomness to the spec. I would add one item of each type to cover full testing opportunities in future

File: spec/request/profile_spec.rb [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      it "should fail to update the profile" do
+        expect {
+          put "/profiles/#{profile.id}", params: { profile: new_attributes }
+        }.not_to change {
+          profile.reload.attributes
+        }
+
+        expect(response).to redirect_to(profiles_path)
+        expect(response).to have_http_status(:redirect)
+        expect(flash[:alert]).to eq("You do not own this profile!")
```
Comment (PiotrMisiurek) [inline spec/request/profile_spec.rb]: You can generalize problem of trying to access/edit not permitted object.

you can change in controller:
```
def set_profile
  @profile = Profile.find(params[:id])
end
```
to 
```
def set_profile
  @profile = current_user.profiles.find(params[:id])
end
```
If a user tries to find not his profile then RecordNotFound will be raised so you don't need `check_owner!` anymore. As trying to access not mine data is not normal user action we don't need to take care of UX here. We can just render generic error or 404 page. You can use `rescue_from RecordNotFound` in `ApplicationController` to achieve that

**Pull request merged**